### PR TITLE
Separate out basic message types from response types

### LINKED ISSUE #53: Separate out basic message types from response types
Currently, there are the message types "UserResp" and "UsersResp". The "UsersResp" has a collection/repeated field called "Items" that is of type "UserResp". For consistency, please create a "User" type that is used as the type in the "Items" field. Similarly, "UserResp" should be updated to have a field called "Item" that is of type "User".

### PR DESCRIPTION
**Pull request by codegold79**
This PR contains fixes for postsvc errors, as well as separate out the user message type from response types.

Closes #53

### CODE REVIEW

File: back/internal/pb/gogen.go [hunk 1]
```diff
@@ -5,5 +5,5 @@ package pb
 //go:generate -command pb-gg protoc -I../../../msgs/proto -I../../../msgs/proto/vendor-extra/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --grpc-gateway_out=logtostderr=true:.
 
 // generate grpc, grpc-gateway
-//go:generate pb-go ../../../msgs/proto/auth.proto ../../../msgs/proto/user.proto
-//go:generate pb-gg ../../../msgs/proto/auth.proto ../../../msgs/proto/user.proto
+//go:generate pb-go ../../../msgs/proto/auth.proto ../../../msgs/proto/user.proto ../../../msgs/proto/posts.proto
+//go:generate pb-gg ../../../msgs/proto/auth.proto ../../../msgs/proto/user.proto ../../../msgs/proto/posts.proto
```
Comment (daved) [inline back/internal/pb/gogen.go]: Please use singular for posts.proto.

File: back/internal/postsvc/internal/postdb/qrypost.go [hunk 2]
```diff
@@ -37,14 +38,19 @@ func (s *PostDB) upsertType(ctx cx, sid string, x *pb.AddTypeReq, y *pb.TypeResp
...
 	}
 
-	y.id = id
-	y.name = x.name
+	intID, err := strconv.Atoi(id.String())
+	if err != nil {
+		return err
+	}
+
+	y.Id = uint32(intID)
```
Comment (daved) [inline back/internal/postsvc/internal/postdb/qrypost.go]: Is the type juggling to/from uint32/string necessary?

File: back/internal/postsvc/internal/postdb/qrypost.go [hunk 3]
```diff
@@ -61,16 +67,21 @@ func (s *PostDB) upsertPost(ctx cx, sid string, x *pb.AddPostReq, y *pb.PostResp
 		return err
 	}
 
-	_, err = stmt.Exec(id, x.typeId, x.title, x.body)
+	_, err = stmt.Exec(id, x.TypeId, x.Title, x.Body)
+
+	if err != nil {
```
Comment (daved) [inline back/internal/postsvc/internal/postdb/qrypost.go]: nit: This could be a single line assignment/conditional (please drop the empty line if not).

Comment (codegold79) [reply back/internal/postsvc/internal/postdb/qrypost.go]: Do you mean to change it to the following?
```
if _, err = stmt.Exec(id, x.TypeId, x.Title, x.Body); err != nil {
    return err
}
```

File: back/internal/postsvc/internal/postdb/qrypost.go (hunk 2 above)
Comment (codegold79) [reply back/internal/postsvc/internal/postdb/qrypost.go]: I might be wrong, but it seems uidgen library returns string IDs, while `post.proto` requires the IDs be uint32. 

Being that I would prefer not to change the uidgen library, should we update post.proto to require string types for IDs?

File: back/internal/pb/gogen.go (hunk 1 above)
Comment (codegold79) [reply back/internal/pb/gogen.go]: Done. I'll push my changes up with the rest of the requested changes.

File: back/internal/postsvc/internal/postdb/qrypost.go (hunk 2 above)
Comment (daved) [reply back/internal/postsvc/internal/postdb/qrypost.go]: If everything else is working for now, please leave these alone and file a new issue to swap the ids in the protos to strings (one issue per proto file, please).

Comment (codegold79) [reply back/internal/postsvc/internal/postdb/qrypost.go]: I've created Issue #58 and Issue #59.

File: back/internal/pb/gogen.go (hunk 1 above)
Comment (codegold79) [reply back/internal/pb/gogen.go]: Done

**Pull request merged**